jina/serve/runtimes/worker/batch_queue.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+from asyncio import Event, Task
+from typing import List, Dict, Any
+
+from jina import DocumentArray
+from jina.serve.executors import BaseExecutor
+from jina.types.request.data import DataRequest
+
+# TODO: Make sure we dont deserialize the request twice
+# TODO: SIGINT/SIGTEM: Flush
```

Comment:
SIGINT SIGTERM is captured by the `WorkerRuntime` and calls `teardown`. So make sure the `teardown` method calls this properly, but do not add specific language about `signal handling` here

Reply:
grpc requests are cancelled before teardown.
even if the queue is flushed, the response wont return so it doesnt work; even when there is no dynamic batching. :(
not going to implement this for now.